scalingo/resource_scalingo_addon.go

```diff
@@ -219,3 +328,25 @@ func waitUntilProvisionned(ctx context.Context, client *scalingo.Client, addon s
 	}
 	return nil
 }
+
+func waitUntilDatabaseFeatureActivated(ctx context.Context, client *scalingo.Client, addon scalingo.Addon, feature string) error {
+	ticker := time.NewTicker(3 * time.Second)
+	defer ticker.Stop()
+	for range ticker.C {
+		db, err := client.DatabaseShow(ctx, addon.AppID, addon.ID)
+		if err != nil {
+			return fmt.Errorf("fail to refresh database metadata: %w", err)
+		}
+		for _, f := range db.Features {
+			if f.Name == feature && f.Status != scalingo.DatabaseFeatureStatusPending {
+				switch f.Status {
+				case scalingo.DatabaseFeatureStatusActivated:
+					return nil
+				case scalingo.DatabaseFeatureStatusFailed:
+					return nil
```

Comment:
question: shouldn't we return an error here?

Reply:
ACK